stl/inc/xutility

```diff
@@ -4963,35 +4958,80 @@ _NODISCARD bool lexicographical_compare(
 #endif // _HAS_CXX17
 
 #ifdef __cpp_lib_concepts
+template <class _Elem1, class _Elem2, class _Cmp>
+struct _Lex_compare_three_way_memcmp_classify_comp {
+    using _Comp = void;
+};
+
+template <class _Elem1, class _Elem2>
+struct _Lex_compare_three_way_memcmp_classify_comp<_Elem1, _Elem2, compare_three_way> {
+    using _Comp = conditional_t<_Lex_compare_memcmp_classify_elements<_Elem1,
+                                    _Elem2> && three_way_comparable_with<const _Elem1&, const _Elem2&>,
+        compare_three_way, void>;
+};
+
+template <class _Ty1, class _Ty2>
+concept _Can_strong_order = requires(const _Ty1& _Left, const _Ty2& _Right) {
+    strong_order(_Left, _Right);
+};

```

Comment:
This appears to duplicate the helper in `<compare>`:
https://github.com/microsoft/STL/blob/d5feb03f320ccaa8e214e3010c40c9c5afd686e2/stl/inc/compare#L663-L666
If they are indeed exactly the same (testing whether we can use the Standard function object), I recommend extracting the helper in `<compare>` out of `namespace _Compare_strong_order_fallback` so it can be used here. This would avoid any quasi-shadowing/confusion.

Occurs for the following concepts too. (I had nested the helper concepts in their namespaces since I didn't expect any further use of them - but didn't intend to prevent wider use.)